### DISCUSSION TITLE
API 추가 데이터 담기 및 파싱 방법

### DISCUSSION THREAD

**Discussion by YuHyun-P** [prgrms-fe-devcourse/FEDC2_2JAVATAYO_Dongyoung#125] [Ideas]
# API 명세서 설명
![image](https://user-images.githubusercontent.com/96400112/173749807-08d18532-4ee3-4971-99d8-2197bfa38394.png)

<br />

### Post
- 모집분야, 연락 방법, 진행 방식, 시작 예정, 예상 기간, 기술 스택, 프로젝트 소개

### User
- 온도

<br />

<details>
  <summary>다른 방법이 있어서 수정될 예정</summary>

### 사용자 정의 JSON 담아서 요청
- title에 저희가 추가로 입력 받는 모집 분야, 연락 방법, 진행 방식, .. 등과 진짜 title을 담아야 합니다. 모든 값은 string으로 보내야 해서 참조값(배열, 객체)라면 아래 skill: JSON.stringify([~]) 처럼 직렬화해서 보내야 응답 받을 때 JSON.parse을 통해 배열로 다시 받을 수 있습니다.
```js
const formData = new FormData();
formData.append(
  "title",
  JSON.stringify({
    title: "JSON 테스트",
    people: "3",
    skill: JSON.stringify(["C/C++", "TypeScript"])
  })
);
formData.append("channelId", "62a55eb0c882bf3a287f9623"); 

postAPI.createPost(formData);
```

### 사용자 정의 JSON 파싱
```js
const getPost = async () => {
  try {
    const response = await postAPI.getPostDetail("62a958533229d934e9b6506e");

    console.log(response.data.title);
    const titleObj = JSON.parse(response.data.title);
    const skills = JSON.parse(titleObj.skill);

    console.log(Array.isArray(skills)); // true

    skills.forEach((skill) => console.log(skill));
  } catch (e) {
    console.error(e);
  }
}
```

<br />

### 파싱
- Post title 필드를 파싱하는 함수 필요
- User fullName 필드를 파싱하는 함수 필요
### 전체 검색 API
- Post title 필드와 User fullName필드를 동시에 검색해서 응답하는 API로 Post title 필드에 추가 데이터를 넣어서 순수 title 검색이 불가능함. → 필터링 함수 필요
  
  ![image](https://user-images.githubusercontent.com/96400112/173751517-e6e6d51d-6d7a-4f8e-b679-590a594cbcba.png)

</details>

**Comment by hwoo3303** [comment]
프로젝트 소개 내용 항목도 없어서 title에 추가해야됩니다. (변수명은 예시와 동일하게 "body"로 하면 될거같아요.) 
게시글 생성, 수정 시 필요하니 혜경님 @Hyevvy 도 참고 부탁드려요 ☺️

![image](https://user-images.githubusercontent.com/15838144/173753305-5ddfa98a-7729-4e1e-93e2-a9abffdaedc4.png)
 
![image](https://user-images.githubusercontent.com/15838144/173753466-b4ecac1c-28da-4cd3-b789-251540faeeb0.png)

**Reply by YuHyun-P** [reply]
네 맞아요!! 위에 코드는 예시 코드라 저희가 사용하는 추가 데이터를 모두 담은 것이 아니고 일부만 담았습니다!
title 필드에 넣을 키 값들은 #72 에 따라서 사용하시면 될 것 같습니다

**Reply by hwoo3303** [reply]
넵 #72 에 추가로 기재해놓을게요!
